lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -660,6 +661,55 @@ int amf_app_handle_pdu_session_response(
 
   return rc;
 }
+/****************************************************************************
+ **                                                                        **
+ ** Name:    convert_ambr()                                                **
+ **                                                                        **
+ ** Description: Converts the session ambr format from                     **
+ **  one defined in create_pdu_session_response to one defined in          **
+ **  pdu_session_estab_accept message                                      **
+ **                                                                        **
+ ** Inputs:  pdu_ambr_response_unit, pdu_ambr_response_value               **
+ **          ambr_unit, ambr_value                                         **
+ **                                                                        **
+ **      Return:   void                                                    **
+ **                                                                        **
+ ***************************************************************************/
+void convert_ambr(
```

Comment:
this conversion does not make too much sense to me. If say the rate is 1.9Mbps, you would be encoding it as 1 Mbps? You should instead define thresholds for the unit changes (e.g., in kbps or in bps) and encode accordingly (e.g., if rate is above X kbps and below Y kbps, use unit Z).

Reply:
add unit test.

Reply:
@ulaskozat , still we won't be able to cover all decimal as finally the values need to be in `uint16`.  `You should instead define thresholds for the unit changes` : Do you mean rounding off to nearest integer?

Reply:
should 1.9 Mbps be converted to 2Mbps?

Reply:
Do you mean if it's 1.9 Mbps, I should use 1900 kbps?

Reply:
Yes, you should encode it such that there is minimum loss of rate information when you convert the value. You should find the absolute thresholds with this objective. It is clear that with 16bits, we have 2^16 * units * (bps/unit). A simple approximation would go to the next unit if the rate to be converted is not covered with smaller unit.

Reply:
@ulaskozat , please review the new function. I've made 65535 as threshold value.